src/ansys/fluent/core/launcher/pim_launcher.py

```diff
@@ -0,0 +1,208 @@
+            The default value is ``None``. Options are ``"meshing"``,
+            ``"pure-meshing"`` and ``"solver"``.
+        py : bool, optional
+            If True, Fluent will run in Python mode. Default is None.
+        gpu : bool, optional
+            If True, Fluent will start with GPU Solver.
+        cwd : str, Optional
+            Working directory for the Fluent client.
+        topy : bool or str, optional
+            A boolean flag to write the equivalent Python journal(s) from the journal(s) passed.
+            Can optionally take the file name of the new python journal file.
+        start_watchdog : bool, optional
+            When ``cleanup_on_exit`` is True, ``start_watchdog`` defaults to True,
+            which means an independent watchdog process is run to ensure
+            that any local GUI-less Fluent sessions started by PyFluent are properly closed (or killed if frozen)
+            when the current Python process ends.
+
+        Returns
+        -------
+        :obj:`~typing.Union` [:class:`Meshing<ansys.fluent.core.session_meshing.Meshing>`, \
+        :class:`~ansys.fluent.core.session_pure_meshing.PureMeshing`, \
+        :class:`~ansys.fluent.core.session_solver.Solver`, \
+        :class:`~ansys.fluent.core.session_solver_icing.SolverIcing`, dict]
+            Session object or configuration dictionary if ``dry_run = True``.
+
+        Raises
+        ------
+        UnexpectedKeywordArgument
+            If an unexpected keyword argument is provided.
+        DockerContainerLaunchNotSupported
+            If a Fluent Docker container launch is not supported.
+
+        Notes
+        -----
+        Job scheduler environments such as SLURM, LSF, PBS, etc. allocates resources / compute nodes.
+        The allocated machines and core counts are queried from the scheduler environment and
+        passed to Fluent.
+        """
+        _process_kwargs(kwargs)
+        del kwargs
+        fluent_launch_mode = _get_fluent_launch_mode(
+            start_container=start_container, container_dict=container_dict
+        )
+        del start_container
+        argvals = locals().copy()
+        _process_invalid_args(dry_run, fluent_launch_mode, argvals)
+        argvals.pop("fluent_launch_mode")
+        args = _get_argvals(argvals, mode)
+        argvals.update(args)
+        for arg_name, arg_values in argvals.items():
+            setattr(self, arg_name, arg_values)
+        self.argvals = argvals
+
+        if self.additional_arguments:
+            logger.warning(
+                "'additional_arguments' option for 'launch_fluent' is currently not supported "
+                "when starting a remote Fluent PyPIM client."
+            )
+
+        if self.start_watchdog:
+            logger.warning(
+                "'start_watchdog' argument for 'launch_fluent' is currently not supported "
+                "when starting a remote Fluent PyPIM client."
+            )
+
+    def __call__(self, **kwargs):
```

Comment:
@hpohekar are you confident about allowing kwargs here? We were going to remove that, right? Even if it appears to work, the issues that I perceive are:
- it mutates the settings for the object, not just for the invocation, so that subsequent invocations are affected
- checks and constraints applied at construction are not mimicked when settings are mutated at invocation

(Glad to see that you've split out into separate files.)

Reply:
@seanpearsonuk Yes, above points are valid and important. We will remove it now. Thank you.

Reply:
Done. Thank you.